lib/spack/spack/detection/common.py

```diff
@@ -299,36 +299,36 @@ def find_windows_compiler_bundled_packages() -> List[str]:
 
 
 class WindowsKitExternalPaths:
-    plat_major_ver = None
-    if sys.platform == "win32":
-        plat_major_ver = str(winOs.windows_version()[0])
-
     @staticmethod
-    def find_windows_kit_roots() -> Optional[str]:
+    def find_windows_kit_roots() -> List[str]:
         """Return Windows kit root, typically %programfiles%\\Windows Kits\\10|11\\"""
         if sys.platform != "win32":
-            return None
+            return []
```

Comment:
Is this called on non-Windows? can we just raise an exception if this is called on non-Windows?

Reply:
Every platform calls it in the process of detecting externals (`Finder.find` on the current develop) since I wanted to avoid overt `if sys.platform==win32` checks there as the more I abstract them, the less ire they seem to draw.